tv_apps/tv-deck/js/tv_deck.js

```diff
@@ -220,209 +160,26 @@
-        this.playingSource.channelIndexHash[channels[i].number] = i;
-        this.playingSource.channels[i] = {
-          channel: channels[i],
-          programs: {},
-        };
-      }
-
-      if (!this.playingChannelNumber) {
-        this.playingChannelNumber = channels[0].number;
-      }
-
-      if (!this.playingChannel) {
-        this._showErrorState();
-        return;
-      }
-
-      this.setHash();
-    }.bind(this), function onerror(error) {
-      alert(error);
-    });
-  };
-
-  proto.setPlayingSource = function td_setPlayingSource(callback) {
-    if (this.playingTuner.tuner.currentSource === this.playingSource.source) {
-      this.setPlayingChannel(callback);
+  proto.setHash = function td_setHash() {
+    if (!this.channelManager.isReady) {
       return;
     }
 
-    this.playingTuner.tuner.setCurrentSource(this.playingSourceType)
-      .then(function() {
-        this.setPlayingChannel(callback);
-      }.bind(this), function() {
-        alert('Source not found');
-      });
-  };
-
-  proto.setPlayingChannel = function td_setPlayingChannel(callback) {
-    this.playingSource.source.setCurrentChannel(this.playingChannelNumber)
-      .then(function() {
-        /* jshint maxlen:false */
-        var newStream = this.playingTuner.tuner.stream;
-        if (this.tvStreamElement.src !== newStream) {
-          this.tvStreamElement.src = newStream;
-          this.tvStreamElement.play();
-        }
-        if (callback) {
-          callback();
-        }
-      }.bind(this), function() {
-        alert('Channel not found');
-      });
-  };
-
-  proto._compareChannel = function td__compareChannel(channelA, channelB) {
-    var dashIndexA = channelA.indexOf('-');
-    var dashIndexB = channelB.indexOf('-');
-    var numberA = (dashIndexA >= 0) ?
-                        channelA.substring(0, dashIndexA) : channelA;
-    var numberB = (dashIndexB >= 0) ?
-                        channelB.substring(0, dashIndexB) : channelB;
-    numberA = parseInt(numberA, 10);
-    numberB = parseInt(numberB, 10);
-
-    if (numberA === numberB) {
-      if (dashIndexA === -1 && dashIndexB === -1) {
-        return 0;
-      } else if (dashIndexA !== -1 && dashIndexB === -1) {
-        return 1;
-      } else if (dashIndexA === -1 && dashIndexB !== -1) {
-        return -1;
-      }
-
-      channelA = channelA.substring(dashIndexA + 1);
-      channelB = channelB.substring(dashIndexB + 1);
-      return this._compareChannel(channelA, channelB);
-    }
-    return numberA - numberB;
-  };
-
-  proto.updateChannelInfo = function td_updateChannelInfo(title, number) {
-    this.channelTitle.textContent = title || this.playingChannel.channel.name;
-    this.channelNumber.textContent = number || this.playingChannelNumber;
-  };
-
-  proto.setHash = function td_setHash() {
-    if (this.currentHash === window.location.hash.substring(1)) {
+    if (this.channelManager.currentHash === window.location.hash.substring(1)) {
```

Comment:
Is it possible to add an extra "#" at `this.channelManager.curentHash`'s implementation? If yes, we don't need to add extra substring(1).